memory_model/src/guest_memory.rs

```diff
@@ -33,17 +33,25 @@ pub enum Error {
 }
 type Result<T> = result::Result<T, Error>;
 
-struct MemoryRegion {
+/// Tracks a mapping of anonymous memory in the current process and the corresponding base address
+/// in the guest's memory space.
+pub struct MemoryRegion {
```

Comment:
I am not super happy for having to make this public :( and the main reason for that is because we are also kind of inserting a function that breaks the pattern of the `guest_memory` implementation. There is a private function called `do_in_region` that gets called when some action needs to be applied on some guest memory region from outside the  module that is agnostic of how the memory region looks.
Ideally, we would have a private function (a more general one) `do_in_all_regions` and then a public one (the more specific one) that calls `fold` on the result. However, this implies quite a few changes and I think it is more important to have that functionality there at this point.

Reply:
I didn't like it too much either but the more we tinker with Firecracker's memory management (and it's likely that we'll be doing that a lot to optimize), the more of `guest_memory` needs to be exposed. So I left it there. Yup, it would take some of tinkering to rework the `fold`ing stuff.